### PR TITLE
Remove PodSecurityPolicy related code except client-go & API type

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?

/kind cleanup
#### What this PR does / why we need it:
PodSecuityPolicy is removed in v1.25.

#### Which issue(s) this PR fixes:
https://github.com/kubernetes/enhancements/issues/5

#### Special notes for your reviewer:
https://github.com/kubernetes/kubernetes/pull/104248#issuecomment-899543857
Some code is not cleaned up for conversion/reading. However, there is no need to keep PSP-related code.

- [x]  cleanup in Kustomize:  https://github.com/kubernetes-sigs/kustomize/pull/4852
- [x] In process: https://github.com/kubernetes/kubernetes/pull/113461
- [x] (TODO in another PR) update sigs.k8s.io/kustomize/kyaml to v0.13.10 to include the https://github.com/kubernetes-sigs/kustomize/pull/4852 (pending on https://github.com/kubernetes-sigs/kustomize/issues/4945)

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by pacoxu** [comment]
> [#104248 (comment)](https://github.com/kubernetes/kubernetes/pull/104248#issuecomment-899543857)
> Some code is not cleaned up for conversion/reading. However, there is no need to keep PSP-related code.

/assign @liggitt

**Comment by liggitt** [comment]
removal of the server-side conversions and registry / storage code lgtm

I think we should keep the client-go and API types _at least_ until 1.24 is out of support, in order to support projects like https://github.com/kubernetes-sigs/pspmigrator/blob/main/pspmutating.go that need the PSP API types in order to assist with migrations to PodSecurity

**Comment by pacoxu** [comment]
> removal of the server-side conversions and registry / storage code lgtm
> 
> 
> 
> I think we should keep the client-go and API types _at least_ until 1.24 is out of support, in order to support projects like https://github.com/kubernetes-sigs/pspmigrator/blob/main/pspmutating.go that need the PSP API types in order to assist with migrations to PodSecurity

Ok. Updated.（remove extensions and keep policy v1beta1 part of client go and api types.

/retest

File: staging/src/k8s.io/api/testdata/v1.24.0/policy.v1beta1.PodSecurityPolicy.json [hunk 1]
```diff
@@ -1,149 +0,0 @@
-{
```
Comment (liggitt) [inline staging/src/k8s.io/api/testdata/v1.24.0/policy.v1beta1.PodSecurityPolicy.json:1]: restore the policy.v1beta1.PodSecurityPolicy files under k8s.io/api/testdata

Comment (pacoxu) [reply staging/src/k8s.io/api/testdata/v1.24.0/policy.v1beta1.PodSecurityPolicy.json]: Done.

**Comment by pacoxu** [comment]
This is just a cleanup. We can merge it in v1.27 if we have no enough time before code freeze.

**Comment by liggitt** [comment]
The changes lgtm, they clean up serving logic around podsecuritypolicies while retaining the API types and clients so client-go can still fetch policy/v1beta1 podsecuritypolicies from an older API server.

I'd like an ack from api-machinery leads (@lavalamp @deads2k) on dropping the extensions/v1beta1 API types/clients (unserved since 1.16 and no longer needed for reading/converting from storage since we no longer serve this API)

/assign @lavalamp @deads2k

**Comment by pacoxu** [comment]
https://github.com/kubernetes/kubernetes/pull/113461 is merged and we can update kustmize to v4.5.8 (next release) once it is released to remove PSP code in it.

**Comment by sftim** [comment]
/retitle Remove PodSecurityPolicy related code except client-go & API type

**Comment by pacoxu** [comment]
Rebased and update `sigs.k8s.io/kustomize/kyaml` to v0.13.10 to include the https://github.com/kubernetes-sigs/kustomize/pull/4852.

**Comment by pacoxu** [comment]
> Rebased and update `sigs.k8s.io/kustomize/kyaml` to v0.13.10 to include the [kubernetes-sigs/kustomize#4852](https://github.com/kubernetes-sigs/kustomize/pull/4852).

The update makes CI failures. I drop the commit.
```
+++ [0104 04:11:09] Running tests without code coverage and with -race
# k8s.io/kubernetes/vendor/sigs.k8s.io/kustomize/api/internal/plugins/fnplugin
vendor/sigs.k8s.io/kustomize/api/internal/plugins/fnplugin/fnplugin.go:60:9: too many return values
	have (*runtimeutil.FunctionSpec, error)
	want (*runtimeutil.FunctionSpec)
```
Maybe we can update the `sigs.k8s.io/kustomize` in an isolated pr to fix it.

**Comment by deads2k** [comment]
glad to see this cleaned up.

/lgtm
/approve

**Comment by pacoxu** [comment]
/test pull-kubernetes-kind-dra
@lavalamp @liggitt do you have time to take a look?

**Pull request merged**